apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/SpannerCommandTests.cs

```diff
@@ -110,67 +111,107 @@ public void CloneWithQueryOptions()
                 using (var reader = command.ExecuteReader())
                 {
-                    // Do nothing.
+                    Assert.True(reader.HasRows);
                 }
-            }, envOptimizerVersion);
+
+                // Optimizer version set through environment variable has higher
+                // precedence than version set through connection.
+                spannerClientMock.Verify(client => client.ExecuteStreamingSql(
+                    It.Is<ExecuteSqlRequest>(request =>
+                        request.QueryOptions.OptimizerVersion == envOptimizerVersion &&
+                        request.QueryOptions.OptimizerStatisticsPackage == envOptimizerStatisticsPackage),
+                    It.IsAny<CallSettings>()));
+            }, envOptimizerVersion, envOptimizerStatisticsPackage);
 
         }
 
         [Fact]
-        public void CommandHasOptimizerVersionSetOnCommand()
+        public void CommandHasQueryOptionsSetOnCommand()
         {
             const string envOptimizerVersion = "2";
-            RunActionWithEnvOptimizerVersion(() =>
+            const string envOptimizerStatisticsPackage = "stats_package_2";
+            RunActionWithEnvQueryOptions(() =>
             {
-                var cmdOptimizerVersion = "3";
-                // Optimizer version set at a command level has higher precedence
-                // than version set through the connection or the environment
-                // variable.
-                Mock<SpannerClient> spannerClientMock = SetupExecuteStreamingSql(cmdOptimizerVersion);
+                const string cmdOptimizerVersion = "3";
+                const string cmdOptimizerStatisticsPackage = "stats_package_3";
+                Mock<SpannerClient> spannerClientMock = SpannerClientHelpers
+                    .CreateMockClient(Logger.DefaultLogger, MockBehavior.Strict);
+                spannerClientMock
+                    .SetupBatchCreateSessionsAsync()
+                    .SetupExecuteStreamingSql();
 
                 const string connOptimizerVersion = "1";
+                const string connOptimizerStatisticsPackage = "stats_package_1";
                 SpannerConnection connection = BuildSpannerConnection(spannerClientMock);
                 var queryOptions = QueryOptions.Empty.WithOptimizerVersion(connOptimizerVersion);
+                queryOptions = queryOptions.WithOptimizerStatisticsPackage(connOptimizerStatisticsPackage);
                 connection.QueryOptions = queryOptions;
 
                 var command = connection.CreateSelectCommand("SELECT * FROM FOO");
                 command.QueryOptions = QueryOptions.Empty.WithOptimizerVersion(cmdOptimizerVersion);
+                command.QueryOptions = command.QueryOptions.WithOptimizerStatisticsPackage(cmdOptimizerStatisticsPackage);
                 using (var reader = command.ExecuteReader())
                 {
-                    // Do nothing.
+                    Assert.True(reader.HasRows);
```

Comment:
Having seen this bit of code (with the validation later) three times now, I suspect a helper method would be useful.